### ISSUE TITLE
Does not work with vanilla iterables

### ISSUE THREAD

**Issue by seprich** [akim-mcmath/chai-iterator#9]
Created a class for nodejs, and a vanilla @@iterator implementation according to the [reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) :

```es6
class DStorage {
//...
[Symbol.iterator]() {
  const ctx = this
  return {
    next: () => {
      if (typeof this._current === 'undefined' || typeof this._current.value === 'undefined') return {done: true}
      this._current = this._current.next
      return {value: this._current.value, done: false}
    },
    _current: ctx.root
  }
}
```

When testing this with chai-iterator the result was:
```
TypeError: iterable[Symbol.iterator] is not a function
```
To overcome this issue it turns out that I have to either 1) rewrite the above implementation in a generator form (`*[Symbol.iterator]`) or 2) I have to add unnecessary extra @@iterator like this:

```es6
class DStorage {
//...
[Symbol.iterator]() {
  const ctx = this
  return {
    next: () => {
      if (typeof this._current === 'undefined' || typeof this._current.value === 'undefined') return {done: true}
      this._current = this._current.next
      return {value: this._current.value, done: false}
    },
    _current: ctx.root,
    [Symbol.iterator]() { return this; }
  }
}
```
The extra `[Symbol.iterator]` within the iterator is not required or even directly mentioned in the iterable protocol reference.

**Comment by akim-mcmath** [comment]
Thanks for letting me know about this. I'm looking into it now.

**Comment by akim-mcmath** [comment]
It's fixed and released as `v1.1.2`. Thanks for this contribution.
